hack/kubevirt.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export KUBEVIRT_TAG="${KUBEVIRT_TAG:-main}"
+
+_base_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+_kubectl="${_base_dir}/_kubevirt/cluster-up/kubectl.sh"
+_kubessh="${_base_dir}/_kubevirt/cluster-up/ssh.sh"
+_action=$1
+shift
+
+function kubevirt::install() {
+    if [[ ! -d ${_base_dir}/_kubevirt ]]; then
+        git clone --depth 1 --branch "${KUBEVIRT_TAG}" https://github.com/kubevirt/kubevirt.git "${_base_dir}"/_kubevirt
+    fi
+    source ${_base_dir}/_kubevirt/cluster-up/hack/common.sh
+}
+
+function kubevirt::up() {
+    make cluster-up -C "${_base_dir}/_kubevirt" && make cluster-sync -C "${_base_dir}"/_kubevirt
+}
+
+function kubevirt::down() {
+     make cluster-down -C "${_base_dir}/_kubevirt"
+}
+
+function kubevirt::kubeconfig() {
+  "${_base_dir}/_kubevirt/cluster-up/kubeconfig.sh"
+}
+
+function kubevirt::registry() {
+  echo "localhost:$("${_base_dir}/_kubevirt/cluster-up/cli.sh" ports registry 2>/dev/null)"
```

Comment:
To be honest it still is rather hard to read. Maybe it is worth to assign the subshell output into a variable and then echo it?

Reply:
ACK done.